src/libstd/sync/condvar.rs

```diff
@@ -60,6 +60,10 @@ impl WaitTimeoutResult {
     ///         break
     ///     }
     /// }
+    /// // Leaving a condition variable's mutex held after we're no longer
+    /// // watching it will block and possibly deadlock notifier threads, so
+    /// // we explicitely drop it.
```

Comment:
The salient point is to release the mutex as soon as possible, not to release it explicitly, so I also find this part misleading. Implicit release when it goes out of scope is just as good as long as it happens as soon as possible.

Reply:
Then should I just make the comment something like `It is sometimes necessary to manually drop the mutex to unlock it as soon as possible. If you need the ressource until the end of the scope, this is not necessary.` and put it once in a new example on the mutex type itself ?

Reply:
Yeah I think Mutex could use some example code that demonstrates a realistic use case requiring explicit drop.